src/bin/vip-config-software-update.js

```diff
@@ -49,57 +50,74 @@ cmd.argv( process.argv, async ( arg: string[], opt ) => {
+		}
 
-	const updateData = await promptForUpdate( app.typeId, updateOptions, softwareSettings );
+		const updateOptions: UpdatePromptOptions = {
+			force: !! opt.force,
+		};
 
-	const hasProcessJob = updateData.component !== 'nodejs';
-	const steps = hasProcessJob ? UPDATE_SOFTWARE_PROGRESS_STEPS : [ UPDATE_SOFTWARE_PROGRESS_STEPS[ 0 ] ];
-	const progressTracker = new ProgressTracker( steps );
+		if ( arg.length > 0 ) {
+			updateOptions.component = arg[ 0 ];
+		}
+		if ( arg.length > 1 ) {
+			updateOptions.version = arg[ 1 ];
+		}
+
+		updateData = await promptForUpdate( app.typeId, updateOptions, softwareSettings );
+
+		const hasProcessJob = updateData.component !== 'nodejs';
+		const steps = hasProcessJob ? UPDATE_SOFTWARE_PROGRESS_STEPS : [ UPDATE_SOFTWARE_PROGRESS_STEPS[ 0 ] ];
+		const progressTracker = new ProgressTracker( steps );
 
-	progressTracker.startPrinting();
-	progressTracker.stepRunning( 'trigger' );
+		progressTracker.startPrinting();
+		progressTracker.stepRunning( 'trigger' );
 
-	const triggerResult = await triggerUpdate( { appId: app.id, envId: env.id, ...updateData } );
-	debug( 'Triggered update with result', triggerResult );
+		const triggerResult = await triggerUpdate( { appId: app.id, envId: env.id, ...updateData } );
+		debug( 'Triggered update with result', triggerResult );
 
-	progressTracker.stepSuccess( 'trigger' );
+		progressTracker.stepSuccess( 'trigger' );
 
-	if ( hasProcessJob ) {
-		const { ok, errorMessage } = await getUpdateResult( app.id, env.id );
+		if ( hasProcessJob ) {
+			const { ok, errorMessage } = await getUpdateResult( app.id, env.id );
 
-		if ( ok ) {
-			progressTracker.stepSuccess( 'process' );
+			if ( ok ) {
+				progressTracker.stepSuccess( 'process' );
+			} else {
+				progressTracker.stepFailed( 'process' );
+			}
+			progressTracker.print();
+			progressTracker.stopPrinting();
+
+			if ( ok ) {
+				console.log( chalk.green( '✓' ) + ' Software update complete' );
+			} else {
+				throw Error( errorMessage );
+			}
 		} else {
-			progressTracker.stepFailed( 'process' );
+			progressTracker.print();
+			progressTracker.stopPrinting();
+			const deploymentsLink = `https://dashboard.wpvip.com/apps/${ app.id }/${ env.uniqueLabel }/deploys`;
+			const message = ` A new build of the application code has been initiated and will be deployed using Node.js v${ updateData.version } when the build is successful\n` +
+				`View the deployments page in VIP Dashboard for progress updates. - ${ deploymentsLink }`;
+			console.log( chalk.green( '✓' ) + message );
 		}
-		progressTracker.print();
-		progressTracker.stopPrinting();
-
-		if ( ok ) {
-			console.log( chalk.green( '✓' ) + ' Software update complete' );
+		await trackEvent( 'config_software_update_success', { ...baseTrackingInfo, ...updateData } );
+	} catch ( error ) {
+		if ( error instanceof UserError ) {
+			await trackEvent( 'config_software_update_success', { ...baseTrackingInfo, ...updateData, user_error: error?.message } );
```

Comment:
This shouldn't be `success` should it?

Reply:
I think it should. We had similar problem with `dev-env` where one o the most common error was "you need to start your dev-env before doing this ..."-kind a thing. Which is not really a failure that we need to fix.

The UserError is analog to 4xx HTTP error. So user did something wrong not the code. We still have that `user_error` code there.

I guess alternatively we could use totally different key for the user_errors? `config_software_update_user_error` maybe? Nothing like that is used in other commands as far as I know, but it might make sense here?

Reply:
Okay, I see what you mean. If this is intended then all good with me!